geopandas/geodataframe.py

```diff
@@ -927,13 +927,13 @@ def iterfeatures(self, na="null", show_bbox=False, drop_id=False):
         if len(properties_cols) > 0:
             # convert to object to get python scalars.
             properties_cols = self[properties_cols]
-            properties = properties_cols.astype(object).values
+            properties = properties_cols.astype(object)
             na_mask = pd.isna(properties_cols).values
 
             if na == "null":
                 properties[na_mask] = None
```

Comment:
`properties` had become a read-only array with CoW enabled (although this might be avoidable from the pandas side?). I delayed the conversion to a numpy array now until after this setitem. Another option would be to copy the array, but this fix seems better (it should avoid the copy; in case the astype resulted in a copy, pandas shouldn't trigger any copy-on-write)